src/python/pants/backend/python/rules/pytest_runner.py

```diff
@@ -210,6 +216,11 @@ async def setup_pytest_for_target(
         "PYTEST_ADDOPTS": " ".join(add_opts),
         "PEX_EXTRA_SYS_PATH": ":".join(prepared_sources.source_roots),
     }
+    test_extra_env = await Get(TestExtraEnv, TestSubsystem, test_subsystem)
```

Comment:
I think this can be a parameter to the rule, which improves performance ever so slightly.